kvmagent/kvmagent/plugins/vm_metadata/rw.py

```
@@ -0,0 +1,678 @@
+
+# ===================================================================
+# Public API – write
+# ===================================================================
+
+def write_metadata(lv_path, payload, lv_size_getter, lv_extend_func,
+                   schema_version=0):
+    """Three-phase atomic write of metadata to an sblk LV.
+
+    Args:
+        lv_path:         ``/dev/{vg}/{vm}_vmmeta``
+        payload:         bytes | str – metadata JSON payload
```

Ruff (0.15.2)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)